readabilipy/__main__.py

```diff
@@ -54,9 +56,15 @@ def main():
     )
 
     args = parser.parse_args()
+    sys.stdin.reconfigure(encoding="utf-8", errors="replace")
+    input_file = sys.stdin
+    output_file = sys.stdout
+    if args.input_file != "-":
+        input_file = open(args.input_file, encoding="utf-8", errors="replace")
+    if args.output_file != "-":
+        output_file = open(args.output_file, "w", encoding="utf-8")
 
-    with open(args.input_file, encoding="utf-8") as h:
-        html = h.read()
+    html = input_file.read()
```

Comment:
If `input_file` *is* a file, it could be closed here to avoid a dangling file handle.

Reply:
Done.